### PR TITLE
Error handling fixes and improvements

### PR DESCRIPTION
**Pull request by ximon18**
Note: Includes https://github.com/NLnetLabs/krill/pull/463.
https://github.com/NLnetLabs/lagosta/pull/94 should be merged to Lagosta master when this PR is merged to Krill v0.8.1-bis.

### CODE REVIEW

**Review by timbru — COMMENTED** [review]
Seems good to merge.

I had just one remark regarding `unwrap()` for read/write locks, but it's not a blocker. The explicit error handling is fine as well.

File: src/daemon/auth/providers/openid_connect/provider.rs [hunk 1]
```diff
@@ -150,36 +154,36 @@ impl OpenIDConnectAuthProvider {
...
-            }
-            Err(err) => {
-                return Err(OpenIDConnectAuthProvider::internal_error(
-                    "Unable to initialize provider connection: Unable to acquire internal lock",
-                    Some(&err.to_string()),
-                ));
-            }
+        let mut conn_guard = self.conn
+            .write()
+            .map_err(|err| OpenIDConnectAuthProvider::internal_error(
```
[the remark this answers is not in the document]
Comment (ximon18) [reply src/daemon/auth/providers/openid_connect/provider.rs:159]: I'm all for consistency and simplicity so I'll change it to match the rest of Krill.

[the remark this answers is not in the document]
Comment (ximon18) [reply src/daemon/auth/providers/openid_connect/provider.rs]: Done.

**Review by timbru — APPROVED** [review]

**Comment by ximon18** [comment]
Agreed with @timbru to merge this PR as the only failure is due to #445 and not related to this PR.

**Pull request merged**